lib/rom/sql/function.rb

```diff
@@ -132,6 +132,22 @@ def cast(expr, db_type = TypeSerializer[:default].call(type))
         Attribute[type].meta(sql_expr: ::Sequel.cast(expr, db_type))
       end
 
+
+      # Add a CASE clause for handling if/then logic
+      #
+      # @example
+      #   users.select { string::case({ "on" => "active" }, "inactive", :status).as(:activated) }
```

Comment:
`:status` should be `status`, i.e. be a method call rather than a symbol. Symbols work too but we don't advertise them since it's Sequel's API we're not aiming to support

Reply:
_I think_ status should be the first argument, it looks more inline with SQL, easier to remember.
In other words, there are two possible calls:
```ruby
case(matching, default_case)
case(expr, matching, default_case)
```

However, `default_case` is optional so we have to rely on a `is_a?(Hash)` check to distinguish these two calls
```ruby
case(matching, default_case)
case(expr, matching)
```

I know it's more work then simply wrapping a ready Sequel's API but we since have a chance to come up with a better design we should try.

Reply:
I noticed that case have one more use case. The matching argument might be a Hash or Array. 

Okay, anyway I modified implementation a little bit, but to be honest - Idea indeed is very good, but the implementation... I don't feel well with the quality of it. I'm always trying to omit and replace fetching args directly from the position in array e.g `args[2]` etc. ;)
Do you have a similar idea in your mind or I'm don't get something?